pkg/realizer/realizer_test.go

```diff
@@ -145,12 +154,12 @@ var _ = Describe("Realize", func() {
 
 			outputFromFirstResource := &templates.Output{Image: "whatever"}
 
-			resourceRealizer.DoCalls(func(ctx context.Context, resource realizer.OwnerResource, blueprintName string, outputs realizer.Outputs, mapper meta.RESTMapper) (templates.Template, *unstructured.Unstructured, *templates.Output, error) {
+			resourceRealizer.DoCalls(func(ctx context.Context, resource realizer.OwnerResource, blueprintName string, outputs realizer.Outputs, mapper meta.RESTMapper) (templates.Reader, *unstructured.Unstructured, *templates.Output, error) {
 				executedResourceOrder = append(executedResourceOrder, resource.Name)
 				Expect(blueprintName).To(Equal("greatest-supply-chain"))
 				if resource.Name == "resource1" {
 					Expect(outputs).To(Equal(realizer.NewOutputs()))
-					template, err := templates.NewModelFromAPI(template1)
+					template, err := templates.NewReaderFromAPI(template1)
```

Comment:
Matching the variable names to the function output can clarify both what's happening and what's the difference between what is now `template1`  and `template`.

```suggestion
					templateReader, err := templates.NewReaderFromAPI(template1)
```